qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -826,6 +826,151 @@ def __new__(
+
+    The lifted gaussian square drag pulse :math:`f'(x)` is defined as:
+
+    .. math::
+
+        f'(x) &= \\begin{cases}\
+            \\text{A} \\times d'(x, \\text{risefall}, \\text{sigma}, \\text{beta})\
+                & x < \\text{risefall}\\\\
+            \\text{A}\
+                & \\text{risefall} \\le x < \\text{risefall} + \\text{width}\\\\
+            \\text{A} \\times \\times d'(\
+                    x - (\\text{risefall} + \\text{width}),\
+                    \\text{risefall},\
+                    \\text{sigma},\
+                    \\text{beta}\
+                )\
+                & \\text{risefall} + \\text{width} \\le x\
+        \\end{cases}\\\\
+
+    where :math:`\\text{A} = \\text{amp} \\times
+    \\exp\\left(i\\times\\text{angle}\\right)`.
+
+    Args:
+        duration: Pulse length in terms of the sampling period `dt`.
+        amp: The amplitude of the DRAG rise and fall and of the square pulse.
+        sigma: A measure of how wide or narrow the DRAG risefall is; see the class
+               docstring for more details.
+        beta: The DRAG correction amplitude.
+        width: The duration of the embedded square pulse.
+        angle: The angle in radians of the complex phase factor uniformly
+            scaling the pulse. Default value 0.
+        risefall_sigma_ratio: The ratio of each risefall duration to sigma.
+        name: Display name for this pulse envelope.
+        limit_amplitude: If ``True``, then limit the amplitude of the
+            waveform to 1. The default is ``True`` and the amplitude is constrained to 1.
+
+    Returns:
+        SymbolicPulse instance.
+
+    Raises:
+        PulseError: When width and risefall_sigma_ratio are both empty or both non-empty.
+    """
+    # Convert risefall_sigma_ratio into width which is defined in OpenPulse spec
+    if width is None and risefall_sigma_ratio is None:
+        raise PulseError(
+            "Either the pulse width or the risefall_sigma_ratio parameter must be specified."
+        )
+    if width is not None and risefall_sigma_ratio is not None:
+        raise PulseError(
+            "Either the pulse width or the risefall_sigma_ratio parameter can be specified"
+            " but not both."
+        )
+    if width is None and risefall_sigma_ratio is not None:
+        width = duration - 2.0 * risefall_sigma_ratio * sigma
+
+    parameters = {"amp": amp, "sigma": sigma, "width": width, "beta": beta, "angle": angle}
+
+    # Prepare symbolic expressions
+    _t, _duration, _amp, _sigma, _beta, _width, _angle = sym.symbols(
+        "t, duration, amp, sigma, beta, width, angle"
+    )
+    _center = _duration / 2
+
+    _sq_t0 = _center - _width / 2
+    _sq_t1 = _center + _width / 2
+
+    _gaussian_ledge = _lifted_gaussian(_t, _sq_t0, -1, _sigma)
+    _gaussian_redge = _lifted_gaussian(_t, _sq_t1, _duration + 1, _sigma)
+    _deriv_ledge = -(_t - _sq_t0) / (_sigma**2) * _gaussian_ledge
```

Comment:
Is it conventional IBM definition? Seems like this is not a perfect derivative term because the real part is the lifted Gaussian but this differential coefficient is from normal Gaussian. I think we can practically ignore this inconsistency and this envelope should be okey if it is identical to the backend. We may want to write a note for this.

Reply:
Yes, this is the conventional IBM definition. The real part is lifted while the imaginary part remains the derivative of the real part (the lifting doesn't affect the derivative since it's a constant). Since the derivative has opposite signs on each end (and goes through 0 in the middle), there is no way to lift it to bring the ends to 0. Perhaps the ends could be tapered to 0 but in practice, IBM has just used the derivative unmodified.

Were you thinking of a note in the doc string for `GaussianSquareDrag`?

Reply:
Thanks. I realized this is formally written in the doc string. I think this should be okey as it is.